### ISSUE TITLE
Migration from 2.1.0 to 2.2.0 fails during dojo.0120_sonarqube_test_and_clean step

### ISSUE THREAD

**Issue by NicolasBi** [DefectDojo/django-DefectDojo#5028]
**Bug description**
Migration command tool from 2.1.0 to 2.2.0 fails (python manage.py migrate) with this error: 
`Applying dojo.0120_sonarqube_test_and_clean...Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.NotNullViolation: column "sonarqube_tool_config_id" contains null values`

It seems that the table dojo_sonarqube_product contains rows with "null" value of "sonarqube_tool_config_id".
Since we don't use Sonarqube , the deletion of all rows fix the problem.

**Steps to reproduce**
1. Migrate from 2.1.0 to 2.2.0
2. Run the command "python manage.py migrate"

**Expected behavior**
The migration process should proceed even with "null" sonarqube_tool_config_id

**Deployment method** *(select with an `X`)*
- [ ] Docker
- [X] Kubernetes
- [ ] GoDojo
- [ ] setup.bash / legacy-setup.bash

**Environment information**
 - Operating System: EKS

**Comment by valentijnscholten** [comment]
Is there a full traceback showing which line in the migration caused it?

**Comment by valentijnscholten** [comment]
we also thought we fixed this through https://github.com/DefectDojo/django-DefectDojo/pull/5013

**Comment by emresaglam** [comment]
I can say the same. Here is my full trace on the same issue from the logs. (Running on docker)

```
minitializer_1   |   Applying dojo.0119_default_group_is_staff... OK
minitializer_1   | Traceback (most recent call last):
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
minitializer_1   |     return self.cursor.execute(sql, params)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/mysql/base.py", line 73, in execute
minitializer_1   |     return self.cursor.execute(query, args)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 206, in execute
minitializer_1   |     res = self._query(query)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 319, in _query
minitializer_1   |     db.query(q)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/connections.py", line 259, in query
minitializer_1   |     _mysql.connection.query(self, query)
minitializer_1   | MySQLdb._exceptions.OperationalError: (1138, 'Invalid use of NULL value')
minitializer_1   |
minitializer_1   | The above exception was the direct cause of the following exception:
minitializer_1   |
minitializer_1   | Traceback (most recent call last):
minitializer_1   |   File "manage.py", line 11, in <module>
minitializer_1   |     execute_from_command_line(sys.argv)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
minitializer_1   |     utility.execute()
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 395, in execute
minitializer_1   |     self.fetch_command(subcommand).run_from_argv(self.argv)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 330, in run_from_argv
minitializer_1   |     self.execute(*args, **cmd_options)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 371, in execute
minitializer_1   |     output = self.handle(*args, **options)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 85, in wrapped
minitializer_1   |     res = handle_func(*args, **kwargs)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 243, in handle
minitializer_1   |     post_migrate_state = executor.migrate(
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 117, in migrate
minitializer_1   |     state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
minitializer_1   |     state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
minitializer_1   |     state = migration.apply(state, schema_editor)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/migrations/migration.py", line 124, in apply
minitializer_1   |     operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/migrations/operations/fields.py", line 236, in database_forwards
minitializer_1   |     schema_editor.alter_field(from_model, from_field, to_field)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 571, in alter_field
minitializer_1   |     self._alter_field(model, old_field, new_field, old_type, new_type,
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 721, in _alter_field
minitializer_1   |     self.execute(
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 142, in execute
minitializer_1   |     cursor.execute(sql, params)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 66, in execute
minitializer_1   |     return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
minitializer_1   |     return executor(sql, params, many, context)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
minitializer_1   |     return self.cursor.execute(sql, params)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
minitializer_1   |     raise dj_exc_value.with_traceback(traceback) from exc_value
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
minitializer_1   |     return self.cursor.execute(sql, params)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/mysql/base.py", line 73, in execute
minitializer_1   |     return self.cursor.execute(query, args)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 206, in execute
minitializer_1   |     res = self._query(query)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 319, in _query
minitializer_1   |     db.query(q)
minitializer_1   |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/connections.py", line 259, in query
minitializer_1   |     _mysql.connection.query(self, query)
minitializer_1   | django.db.utils.OperationalError: (1138, 'Invalid use of NULL value')
```

**Comment by valentijnscholten** [comment]
@kiblik

**Comment by kiblik** [comment]
I see. I'm going to try to write a hot-fix.

**Comment by kiblik** [comment]
#5036

I would like to write some test for it as well

**Comment by valentijnscholten** [comment]
fixed in 2.2.1
